core/state/snapshot/iterator.go

```diff
@@ -125,8 +125,10 @@ func (it *diffAccountIterator) Account() []byte {
 	blob, ok := it.layer.accountData[it.curHash]
 	if !ok {
 		if _, ok := it.layer.destructSet[it.curHash]; ok {
+			it.layer.lock.RUnlock()
```

Comment:
Good catch. But please remove the other unlocks from before panics. The panics are there to catch programming errors, but there's no way to recover from them. At that point the program is going down hard.

Reply:
I think it does no harm to add Unlock before panic since we cannot guarantee the panic will not be recovered in the future design.

Reply:
It makes code harder to understand and makes you wonder why it is there if you panic. Re recover, no, it won't. Panics are not meant to be used as an error handling. If your code hits this path, all the internal state is irreversibly corrupted. There's no way out, there's nothing meaningful that can be done.

Reply:
Okay, I see. Should I also remove the Unlock before panic in `go-ethereum/core/state/snapshot/snapshot.go`?